src/Compilers/VisualBasic/Portable/Syntax/SyntaxUtilities.vb

```diff
@@ -43,12 +44,43 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                      SyntaxKind.DescendingOrdering
                     Return DirectCast(newLambda, OrderingSyntax).Expression
 
+                Case SyntaxKind.SelectClause
+                    Dim oldSelect = DirectCast(oldLambda, SelectClauseSyntax)
+                    Return DirectCast(newLambda, SelectClauseSyntax).Variables.First
```

Comment:
It is not obvious why returning the first variable is the right thing to do here.